### ISSUE TITLE
Training the Siamese Network on triplets

### ISSUE THREAD

**Issue by pskrunner14** [PacktPublishing/Java-Machine-Learning-for-Computer-Vision#3]
Hi I could'nt find any implementation of training the siamese net on triplet loss function in the code as explained in the post, would'nt it be necessary to train the Siamese network using triplet loss in order to increase the distance between face embeddings of different peoples' faces and decrease distance between embeddings of the same person's face? The current model performs poorly when tested on new faces or even face embeddings of similar looking people in the database.

**Comment by klevis** [comment]
Hi 
You can find how to train in the java class https://github.com/PacktPublishing/Java-Machine-Learning-for-Computer-Vision/blob/master/FaceRecognition/src/main/java/ramo/klevis/ml/recogntion/face/FaceNetSmallV2Model.java , please un-comment the last line on **conf** method. The lost layer is CenterLossOutputLayer which is almost identical to triplet lost just a bit faster.
About the accuracy I addressed in the last section of post : http://ramok.tech/2018/09/12/java-home-made-face-recognition-application/#Limitations
Basically the weights from Keras to DL4J are not still 100% tuned due to some difference in internal workings of respective frameworks and please keep in mind that even the Keras weights are not production one, real system will be much much bigger. Anyway the model could be improved and trained further so feel free to try it. 
Please stay tuned as the code it will continually improved to a state of the art accuracy eventually.

**Comment by bewithme** [comment]
you just tell us un-comment the last line on conf method. but can you give the full example of training FaceNetSmallV2Model?  I don't know what data set should be used to train FaceNetSmallV2Model.
